CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/utility/SelectAllPluginNGTest.java

```diff
@@ -0,0 +1,328 @@
+import au.gov.asd.tac.constellation.plugins.PluginException;
+import au.gov.asd.tac.constellation.plugins.PluginExecution;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * Cases of Edit tested: empty graph with visual schema, empty graph with bare
+ * schema, empty graph visual schema with attributes, visual graph with two vx
+ * and attributes, visual graph with two vx and two tx.
+ *
+ *
+ * @author aldebaran30701
+ */
+public class SelectAllPluginNGTest {
+
+    private Graph graph;
+    private int selectedV, selectedT;
+    private int vxId1, vxId2, txId1, txId2;
+
+    public SelectAllPluginNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of edit method, of class SelectAllPlugin. No attributes added so
+     * should throw a pluginexception
+     */
+    @Test(expectedExceptions = PluginException.class)
+    public void testEdit() throws Exception {
+        System.out.println("select all edit with nothing on graph");
+        // Open a new graph
+        graph = new DualGraph(SchemaFactoryUtilities.getSchemaFactory(VisualSchemaFactory.VISUAL_SCHEMA_ID).createSchema());
+        int vxCount = 0;
+        int txCount = 0;
+
+        // run select all plugin
+        PluginExecution.withPlugin(new SelectAllPlugin()).executeNow(graph);
+
+        // Verify no extra elements are added
+        final ReadableGraph rg = graph.getReadableGraph();
+        try {
+            assertEquals(rg.getVertexCount(), vxCount);
+            assertEquals(rg.getTransactionCount(), txCount);
+        } finally {
+            rg.close();
+        }
+    }
+
+    /**
+     * Test of edit method, of class SelectAllPlugin. select all edit with only
+     * attributes on graph
+     */
+    @Test
+    public void testEdit0() throws Exception {
+        System.out.println("select all edit with nothing on graph but attributes");
+        // Open a new graph
+        graph = new DualGraph(SchemaFactoryUtilities.getSchemaFactory(VisualSchemaFactory.VISUAL_SCHEMA_ID).createSchema());
+        int vxCount = 0;
+        int txCount = 0;
+        final WritableGraph wg = graph.getWritableGraph("TEST", true);
+        try {
+            // Create Selected Attributes
+            selectedV = VisualConcept.VertexAttribute.SELECTED.ensure(wg);
+            if (selectedV == Graph.NOT_FOUND) {
```

Comment:
Given `ensure` creates the attribute if it doesn't exist, this if will always be false

Reply:
I just copied this from another test and thought it'd be useful to keep it in. I can get rid of it for clarity.